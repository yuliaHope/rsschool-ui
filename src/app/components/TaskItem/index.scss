@import '../../../styles/bootswatch/_variables';
@import '../../../styles/variables';

.three-in-row {
  flex-basis: 25% !important;
}

.header-alignment {
  display: flex;
  justify-content: space-between;
}

.footer-form {
  background-color: #fff ;
  border-top: none ;
}
