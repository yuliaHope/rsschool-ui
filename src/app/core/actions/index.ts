export * from './schedule';
export * from './user';
export * from './courses';
export * from './assignments';
