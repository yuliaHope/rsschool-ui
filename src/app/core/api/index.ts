export * from './schedule';
export * from './session';
export * from './user';
export * from './courses';
export * from './assignments';
