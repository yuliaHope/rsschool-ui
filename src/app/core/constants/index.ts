export * from './schedule';
export * from './courses';
export * from './user';
export * from './assignments';
