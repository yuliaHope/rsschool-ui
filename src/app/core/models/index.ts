export * from './events';
export * from './stage';
export * from './user';
export * from './course';
export * from './assignments';
export * from './courseMentor';
export * from './courseStudent';
