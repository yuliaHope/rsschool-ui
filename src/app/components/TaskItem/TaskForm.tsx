import * as React from 'react';
import { IAssignmentModel } from 'core/models';

type Props = {
    status: string;
    assignment: IAssignmentModel;
    fetchAssignmentSolution: (assignment: IAssignmentModel) => void;
};

class TaskForm extends React.Component<Props> {
    private studentRepo: React.RefObject<HTMLInputElement>;
    private studentComment: React.RefObject<HTMLTextAreaElement>;
    constructor(props: Props) {
        super(props);
        this.studentRepo = React.createRef();
        this.studentComment = React.createRef();
    }

    SendStudentTask(assignment: IAssignmentModel) {
        const { fetchAssignmentSolution } = this.props;
        const studentRepoValue: string = this.studentRepo.current!.value.replace(/\s/g, '');
        if (studentRepoValue) {
            fetchAssignmentSolution(assignment);
        } else {
            alert('wrong repo');
            this.studentRepo.current!.value = '';
        }
    }

    render() {
        const { status } = this.props;
        if (status !== 'MissedDeadline' && status !== 'Checked' && status !== 'ReadyForReview') {
            const { assignment } = this.props;
            return (
                <small className="text-muted">
                    <form
                        onSubmit={ev => {
                            ev.preventDefault();
                            this.SendStudentTask(assignment);
                        }}
                    >
                        <div className="form-group">
                            <label>Choose repo</label>
                            <input
                                className="form-control form-control-sm"
                                type="text"
                                ref={this.studentRepo}
                                placeholder="Enter link"
                                pattern="https?://github.com/.+"
                            />
                        </div>
                        <div className="form-group">
                            <label>Comments</label>
                            <textarea
                                className="form-control form-control-sm"
                                ref={this.studentComment}
                                placeholder="Write comments here"
                            />
                        </div>
                        <button type="submit" className="btn btn-primary btn-sm">
                            Submit
                        </button>
                    </form>
                </small>
            );
        } else {
            return <React.Fragment />;
        }
    }
}

export default TaskForm;
