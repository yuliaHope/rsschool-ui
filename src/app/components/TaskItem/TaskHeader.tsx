import * as React from 'react';
import { classNames } from '../../core/styles';
import { AssignmentStatus } from '../../core/models';
const cn = classNames(require('./index.scss'));

type Props = {
    status: string;
    score: number;
};
export const TaskHeader = ({ status, score }: Props) => {
    let headerTxt;
    let mark;
    let headerClassNames;
    if (status === AssignmentStatus.Assigned) {
        headerTxt = 'Not submitted yet!';
        mark = '';
        headerClassNames = 'bg-warning';
    } else if (status === AssignmentStatus.MissedDeadline) {
        headerTxt = 'The deadline has passed!';
        mark = '';
        headerClassNames = 'bg-danger';
    } else if (status === AssignmentStatus.Checked) {
        headerTxt = 'Done!';
        mark = score;
        headerClassNames = mark >= 100 ? 'bg-success' : 'bg-warning';
    } else {
        headerTxt = status;
        mark = '';
        headerClassNames = '';
    }

    const colorizeText = headerClassNames ? 'text-white' : '';

    return (
        <div className={`card-header ${headerClassNames} ${colorizeText} ${cn('header-alignment')}`}>
            <span className="card-header-text"> {headerTxt} </span>
            {mark ? <span className="card-header-text">{mark}%</span> : null}
        </div>
    );
};
