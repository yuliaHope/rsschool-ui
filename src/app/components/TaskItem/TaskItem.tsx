import * as React from 'react';
import { IAssignmentModel } from 'core/models';
import { classNames } from '../../core/styles';
import TaskForm from './TaskForm';
import { TaskHeader } from './TaskHeader';

const cn = classNames(require('./index.scss'));

type Props = {
    assignment: IAssignmentModel;
    fetchAssignmentSolution: (assignment: IAssignmentModel) => void;
};
export const TaskItem = ({ assignment, fetchAssignmentSolution }: Props) => {
    return (
        <div className={cn('three-in-row', 'card', 'bg-secondary', 'mb-3')}>
            <TaskHeader status={assignment.status} score={assignment.score} />
            <div className="card-body">
                <h5 className="card-title">{assignment.title}</h5>
                <p className="card-text">{assignment.assignmentRepo}</p>
            </div>
            {assignment.status !== 'Checked' && (
                <div className={`card-footer ${cn('footer-form')}`}>
                    <TaskForm
                        status={assignment.status}
                        assignment={assignment}
                        fetchAssignmentSolution={fetchAssignmentSolution}
                    />
                </div>
            )}
        </div>
    );
};
