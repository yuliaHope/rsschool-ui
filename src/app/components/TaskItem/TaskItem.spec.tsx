import * as React from 'react';
import { shallow } from 'enzyme';
import { TaskItem } from './TaskItem';
import { AssignmentStatus } from '../../core/models';

describe('Task Item', () => {
    const props = {
        assignment: {
            _id: '_id',
            assignmentRepo: 'assignmentRepo',
            checkDate: 123,
            completeDate: 123,
            courseId: 'courseId',
            deadlineDate: 123,
            mentorComment: 'mentorComment',
            mentorId: 'mentorId',
            score: 0,
            status: AssignmentStatus.Assigned,
            studentComment: 'studentComment',
            studentId: 'studentId',
            taskId: 'taskId',
            title: 'title',
        },
        fetchAssignmentSolution: jest.fn(),
    };
    describe('render item', () => {
        it('Assigned status', () => {
            const output = shallow(<TaskItem {...props} />);
            expect(output).toMatchSnapshot();
        });
        it('in case status is Checked', () => {
            const nextProps = {
                ...props,
                assignment: {
                    ...props.assignment,
                    status: AssignmentStatus.Checked,
                },
            };
            const output = shallow(<TaskItem {...nextProps} />);
            expect(output).toMatchSnapshot();
        });
    });
});
