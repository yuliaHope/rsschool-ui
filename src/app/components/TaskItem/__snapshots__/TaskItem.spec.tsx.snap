// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task Item render item Assigned status 1`] = `
<div
  className="three-in-row card bg-secondary mb-3"
>
  <Component
    score={0}
    status="Assigned"
  />
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      title
    </h5>
    <p
      className="card-text"
    >
      assignmentRepo
    </p>
  </div>
  <div
    className="card-footer footer-form"
  >
    <TaskForm
      assignment={
        Object {
          "_id": "_id",
          "assignmentRepo": "assignmentRepo",
          "checkDate": 123,
          "completeDate": 123,
          "courseId": "courseId",
          "deadlineDate": 123,
          "mentorComment": "mentorComment",
          "mentorId": "mentorId",
          "score": 0,
          "status": "Assigned",
          "studentComment": "studentComment",
          "studentId": "studentId",
          "taskId": "taskId",
          "title": "title",
        }
      }
      fetchAssignmentSolution={[MockFunction]}
      status="Assigned"
    />
  </div>
</div>
`;

exports[`Task Item render item in case status is Checked 1`] = `
<div
  className="three-in-row card bg-secondary mb-3"
>
  <Component
    score={0}
    status="Checked"
  />
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      title
    </h5>
    <p
      className="card-text"
    >
      assignmentRepo
    </p>
  </div>
</div>
`;
